### ISSUE TITLE
Indexer Connector - Verify occurrence of error 409

### ISSUE THREAD

**Issue by sebasfalcone** [wazuh/wazuh#27259]
## Description
During the release testing stage for 4.10.0, the following error appears in the logs:
```console
2024/12/06 14:17:06 indexer-connector: ERROR: HTTP response code said error, status code: 409.
```

We should look into why this error happens, its impact and a solution, if needed.

## DoD
- [x] Determined root cause
- [x] Determined impact
- [x] Proposed solution

**Comment by sebasfalcone** [comment]
## Analysis
I will start by attempting to reproduce this issue

**Comment by sebasfalcone** [comment]
# Root Cause
The error can be replicated by:

1. Creating the same index twice
2. Documents in a delete-by-query were updated while removing them ([see ref](https://discuss.elastic.co/t/elasticsearch-delete-by-query-409-version-conflict/174150)):

## Creating the same index
```console
curl -k -u admin:rmaMWnL?rJCa+KMp8fziovi95*cCkMJ4 -XPUT "https://localhost:9200/testing/_create/100" -H 'Content-Type: application/json' -d '
{
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 1
  }
}'

{
  "_index": "testing",
  "_id": "100",
  "_version": 1,
  "result": "created",
  "_shards": {
    "total": 2,
    "successful": 1,
    "failed": 0
  },
  "_seq_no": 0,
  "_primary_term": 1
}
```

```console
curl -k -u admin:rmaMWnL?rJCa+KMp8fziovi95*cCkMJ4 -XPUT "https://localhost:9200/testing/_create/100" -H 'Content-Type: application/json' -d '
{
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 1
  }
}'

{
  "error": {
    "root_cause": [
      {
        "type": "version_conflict_engine_exception",
        "reason": "[100]: version conflict, document already exists (current version [1])",
        "index": "testing",
        "shard": "0",
        "index_uuid": "rAgWDm_RRT6uZnW7JiOR9A"
      }
    ],
    "type": "version_conflict_engine_exception",
    "reason": "[100]: version conflict, document already exists (current version [1])",
    "index": "testing",
    "shard": "0",
    "index_uuid": "rAgWDm_RRT6uZnW7JiOR9A"
  },
  "status": 409
}
```

## Delete by query
Using this python script
```py
import threading
from time import sleep
from opensearchpy import OpenSearch
import random
import urllib3

# Configuration
OPENSEARCH_HOST = "http://localhost:9200"
INDEX_NAME = "testing"
NUM_THREADS = 5
DOCS_PER_THREAD = 10000
USERNAME = "admin"
PASSWORD = "admin"

urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

# Initialize OpenSearch client
client = OpenSearch(
    hosts=[OPENSEARCH_HOST],
    http_compress=True,
    http_auth=(USERNAME, PASSWORD),
    use_ssl=True,
    verify_certs=False 
)

def insert_documents(thread_id):
    """Function to insert documents into the OpenSearch index."""
    for i in range(DOCS_PER_THREAD):
        # Overlapping IDs are generated using thread_id and a random number
        doc_id = random.randint(1, 1000)  # This will create overlapping IDs
        document = {
            "title": f"Document {i} from thread {thread_id}",
            "content": f"This is the content of document {i} from thread {thread_id}.",
            "timestamp": "2024-12-11T00:00:00Z"
        }
        try:
            response = client.index(
                index=INDEX_NAME,
                id=doc_id,
                body=document
            )
            response_code = response.get('_shards', {}).get('successful', 0)
            print(f"Thread {thread_id}: Indexed document {doc_id}, Response Code: {response_code}")
        except Exception as e:
            print(f"Thread {thread_id}: Error indexing document {doc_id}: {e}")
        
        # Optional: simulate workload
        sleep(0.01)

def main():
    """Main function to create threads and insert documents."""
    threads = []

    for thread_id in range(NUM_THREADS):
        thread = threading.Thread(target=insert_documents, args=(thread_id,))
        threads.append(thread)
        thread.start()

    for thread in threads:
        thread.join()

    print("All threads completed.")

if __name__ == "__main__":
    main()
```

And performing a delete_by_query operation while it was running:
```json
{
  "took": 89,
  "timed_out": false,
  "total": 1723,
  "deleted": 1694,
  "batches": 2,
  "version_conflicts": 29,
  "noops": 0,
  "retries": {
    "bulk": 0,
    "search": 0
  },
  "throttled_millis": 0,
  "requests_per_second": -1,
  "throttled_until_millis": 0,
  "failures": [
    {
      "index": "testing",
      "id": "241",
      "cause": {
        "type": "version_conflict_engine_exception",
        "reason": "[241]: version conflict, required seqNo [98404], primary term [1]. current document has seqNo [98847] and primary term [1]",
        "index": "testing",
        "shard": "0",
        "index_uuid": "KAdYAFRYSZmvJ1YQCjzpgA"
      },
      "status": 409
    },
    {
      "index": "testing",
      "id": "254",
      "cause": {
        "type": "version_conflict_engine_exception",
        "reason": "[254]: version conflict, required seqNo [98420], primary term [1]. current document has seqNo [98858] and primary term [1]",
        "index": "testing",
        "shard": "0",
        "index_uuid": "KAdYAFRYSZmvJ1YQCjzpgA"
      },
      "status": 409
    },
```

**Comment by sebasfalcone** [comment]
## Impact
This issue will prevent the `delete_by_query` operation from running, potentially leaving us in an unknown state

**Comment by sebasfalcone** [comment]
## Impact
The failed `delete_by_query` event, will be re-processed in the next dispatch operation, removing the information that was being indexed (and should be removed)

**Comment by sebasfalcone** [comment]
## Solution
I the problem is related to how the cluster components communicate, we have no proper way to coordinate these operations. This issue will be resolved in 5.0.0
